FWCore/Framework/interface/EDConsumerBase.h

```diff
@@ -247,6 +255,8 @@ namespace edm {
     edm::SoATuple<ESTokenLookupInfo, ESProxyIndex> m_esTokenInfo;
     std::array<std::vector<ESProxyIndex>, static_cast<unsigned int>(edm::Transition::NumberOfTransitions)>
         esItemsToGetFromTransition_;
+    std::array<std::vector<ESRecordIndex>, static_cast<unsigned int>(edm::Transition::NumberOfTransitions)>
+        esRecordsToGetFromTransition_;
```

Comment:
I'm not sure which PR will go in first, this one or the ProcessBlock PR. Whichever goes second is going to need a little work in EDConsumerBase to work properly.